services/src/main/java/io/opentelemetry/android/internal/services/network/detector/NetworkDetector.kt

```diff
@@ -21,22 +27,79 @@ interface NetworkDetector {
     companion object {
         @JvmStatic
         fun create(context: Context): NetworkDetector {
-            // TODO: Use ServiceManager to get the ConnectivityManager or similar (not yet managed/abstracted)
             val connectivityManager =
                 context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
-            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
-                // TODO: Use ServiceManager to get the TelephonyManager or similar (not yet managed/abstracted)
-                val telephonyManager =
-                    context.getSystemService(Context.TELEPHONY_SERVICE) as TelephonyManager
-                val carrierFinder = CarrierFinder(telephonyManager)
-                return PostApi28NetworkDetector(
-                    connectivityManager,
-                    telephonyManager,
-                    carrierFinder,
-                    context,
-                )
+            val telephonyManager =
+                context.getSystemService(Context.TELEPHONY_SERVICE) as TelephonyManager
+            val carrierFinder = CarrierFinder(context, telephonyManager)
+            val subTypeFinder = SubTypeFinder(context, telephonyManager)
+
+            return object : NetworkDetector {
```

Comment:
Consider separating this out from the source file containing the interface, given the complexity has grown somewhat